### ISSUE TITLE
Stopping a timer causes a crash

### ISSUE THREAD

**Issue by ConnorRigby** [membraneframework/membrane-core#266]
In my Sink Element i have: 
```elixir
  @impl true
  def handle_tick(:demand_timer, _ctx, state) do
    Logger.info("alert complete")
    actions = [
      demand: :input,
      stop_timer: :demand_timer,
      notify: {:alert_completed, state.current_alert}
    ]
    {{:ok, actions}, %{state | timer_started: false, current_alert: nil}}
  end
```

Which causes the following crash:

```
[error] GenServer #PID<0.4982.0> terminating
** (MatchError) no match of right hand side value: nil
    (membrane_core 0.5.2) lib/membrane/core/timer.ex:54: Membrane.Core.Timer.tick/1
    (elixir 1.10.3) lib/kernel.ex:2351: anonymous fn/2 in Kernel.update_in/3
    (elixir 1.10.3) lib/map.ex:837: Map.get_and_update/3
    (elixir 1.10.3) lib/access.ex:346: Access.get_and_update/3
    (elixir 1.10.3) lib/kernel.ex:2351: Kernel.update_in/3
    (membrane_core 0.5.2) lib/membrane/core/element/timer_controller.ex:66: Membrane.Core.Element.TimerController.handle_tick/2
    (membrane_core 0.5.2) lib/membrane/core/element/message_dispatcher.ex:36: Membrane.Core.Element.MessageDispatcher.handle_message/3
    (stdlib 3.12.1) gen_server.erl:637: :gen_server.try_dispatch/4
    (stdlib 3.12.1) gen_server.erl:711: :gen_server.handle_msg/6
    (stdlib 3.12.1) proc_lib.erl:249: :proc_lib.init_p_do_apply/3
Last message: {Membrane.Core.Message, :timer_tick, :demand_timer, []}
State: %Membrane.Core.Element.State{controlling_pid: #PID<0.4938.0>, delayed_demands: %{}, internal_state: %{alert_duration: 10000000000, camera: #KnowFalls.Cameras.Camera<__meta__: #Ecto.Schema.Metadata<:loaded, "cameras">, alerts: #Ecto.Association.NotLoaded<association :alerts is not loaded>, id: 1, inserted_at: ~N[2020-07-28 14:36:44], name: "AXIS M1065-LW - ACCC8EC8A54C H\\.264 1._rtsp._tcp.local", serialnbr: "ACCC8EC8A54C", updated_at: ~N[2020-07-28 14:36:44], uri: "rtsp://192.168.1.137:554/axis-media/media.amp/", user: #KnowFalls.Accounts.User<__meta__: #Ecto.Schema.Metadata<:loaded, "users">, cameras: #Ecto.Association.NotLoaded<association :cameras is not loaded>, ...>, user_id: 1, username: "root", ...>, caps: nil, current_alert: %KnowFalls.Alerts.Alert{__meta__: #Ecto.Schema.Metadata<:loaded, "alerts">, camera: #Ecto.Association.NotLoaded<association :camera is not loaded>, camera_id: 1, id: 16, inserted_at: ~N[2020-07-28 16:12:19], notes: nil, processing_completed_at: nil, processing_result: nil, processing_started_at: nil, set: #Ecto.Association.NotLoaded<association :set is not loaded>, set_id: 2, updated_at: ~N[2020-07-28 16:12:19]}, timer_started: true}, module: KnowFalls.AxisCameraPipeline.AlertProcessor, name: {:alert_processor, #Reference<0.189328267.2396520455.163902>}, pads: %{data: %{input: %Membrane.Pad.Data{accepted_caps: :any, availability: :always, bin?: false, caps: nil, demand: 1, demand_unit: :buffers, direction: :input, end_of_stream?: false, input_buf: %Membrane.Core.InputBuffer{current_size: 0, demand: -2, metric: Membrane.Buffer.Metric.Count, min_demand: 10, name: {:alert_processor, #Reference<0.189328267.2396520455.163902>}, preferred_size: 40, q: #Qex<[]>, toilet?: false, toilet_props: %{fail: 160, warn: 80}}, mode: :pull, name: :input, options: nil, other_demand_unit: nil, other_ref: :output, pid: #PID<0.4987.0>, ref: :input, start_of_stream?: true, sticky_messages: []}}, dynamic_currently_linking: [], info: %{}}, parent_monitor: #Reference<0.189328267.2396520455.163904>, playback: %Membrane.Core.Playback{async_state_change: false, pending_state: nil, state: :playing, target_state: :playing}, playback_buffer: %Membrane.Core.Element.PlaybackBuffer{q: #Qex<[]>}, synchronization: %{clock: nil, latency: 0, parent_clock: #PID<0.4939.0>, stream_sync: :membrane_no_sync, timers: %{demand_timer: %Membrane.Core.Timer{clock: #PID<0.4939.0>, id: :demand_timer, init_time: -576459688866892175, interval: 10000000000, ratio: 1, time_passed: 10000000000, timer_ref: #Reference<0.189328267.2396520450.191913>}}}, type: :sink, watcher: #PID<0.4938.0>}
[error] GenServer #PID<0.4982.0> terminating
** (MatchError) no match of right hand side value: nil
    (membrane_core 0.5.2) lib/membrane/core/timer.ex:54: Membrane.Core.Timer.tick/1
    (elixir 1.10.3) lib/kernel.ex:2351: anonymous fn/2 in Kernel.update_in/3
    (elixir 1.10.3) lib/map.ex:837: Map.get_and_update/3
    (elixir 1.10.3) lib/access.ex:346: Access.get_and_update/3
    (elixir 1.10.3) lib/kernel.ex:2351: Kernel.update_in/3
    (membrane_core 0.5.2) lib/membrane/core/element/timer_controller.ex:66: Membrane.Core.Element.TimerController.handle_tick/2
    (membrane_core 0.5.2) lib/membrane/core/element/message_dispatcher.ex:36: Membrane.Core.Element.MessageDispatcher.handle_message/3
    (stdlib 3.12.1) gen_server.erl:637: :gen_server.try_dispatch/4
    (stdlib 3.12.1) gen_server.erl:711: :gen_server.handle_msg/6
    (stdlib 3.12.1) proc_lib.erl:249: :proc_lib.init_p_do_apply/3
Last message: {Membrane.Core.Message, :timer_tick, :demand_timer, []}
State: %Membrane.Core.Element.State{controlling_pid: #PID<0.4938.0>, delayed_demands: %{}, internal_state: %{alert_duration: 10000000000, camera: #KnowFalls.Cameras.Camera<__meta__: #Ecto.Schema.Metadata<:loaded, "cameras">, alerts: #Ecto.Association.NotLoaded<association :alerts is not loaded>, id: 1, inserted_at: ~N[2020-07-28 14:36:44], name: "AXIS M1065-LW - ACCC8EC8A54C H\\.264 1._rtsp._tcp.local", serialnbr: "ACCC8EC8A54C", updated_at: ~N[2020-07-28 14:36:44], uri: "rtsp://192.168.1.137:554/axis-media/media.amp/", user: #KnowFalls.Accounts.User<__meta__: #Ecto.Schema.Metadata<:loaded, "users">, cameras: #Ecto.Association.NotLoaded<association :cameras is not loaded>, ...>, user_id: 1, username: "root", ...>, caps: nil, current_alert: %KnowFalls.Alerts.Alert{__meta__: #Ecto.Schema.Metadata<:loaded, "alerts">, camera: #Ecto.Association.NotLoaded<association :camera is not loaded>, camera_id: 1, id: 16, inserted_at: ~N[2020-07-28 16:12:19], notes: nil, processing_completed_at: nil, processing_result: nil, processing_started_at: nil, set: #Ecto.Association.NotLoaded<association :set is not loaded>, set_id: 2, updated_at: ~N[2020-07-28 16:12:19]}, timer_started: true}, module: KnowFalls.AxisCameraPipeline.AlertProcessor, name: {:alert_processor, #Reference<0.189328267.2396520455.163902>}, pads: %{data: %{input: %Membrane.Pad.Data{accepted_caps: :any, availability: :always, bin?: false, caps: nil, demand: 1, demand_unit: :buffers, direction: :input, end_of_stream?: false, input_buf: %Membrane.Core.InputBuffer{current_size: 0, demand: -2, metric: Membrane.Buffer.Metric.Count, min_demand: 10, name: {:alert_processor, #Reference<0.189328267.2396520455.163902>}, preferred_size: 40, q: #Qex<[]>, toilet?: false, toilet_props: %{fail: 160, warn: 80}}, mode: :pull, name: :input, options: nil, other_demand_unit: nil, other_ref: :output, pid: #PID<0.4987.0>, ref: :input, start_of_stream?: true, sticky_messages: []}}, dynamic_currently_linking: [], info: %{}}, parent_monitor: #Reference<0.189328267.2396520455.163904>, playback: %Membrane.Core.Playback{async_state_change: false, pending_state: nil, state: :playing, target_state: :playing}, playback_buffer: %Membrane.Core.Element.PlaybackBuffer{q: #Qex<[]>}, synchronization: %{clock: nil, latency: 0, parent_clock: #PID<0.4939.0>, stream_sync: :membrane_no_sync, timers: %{demand_timer: %Membrane.Core.Timer{clock: #PID<0.4939.0>, id: :demand_timer, init_time: -576459688866892175, interval: 10000000000, ratio: 1, time_passed: 10000000000, timer_ref: #Reference<0.189328267.2396520450.191913>}}}, type: :sink, watcher: #PID<0.4938.0>}
[error] GenServer #PID<0.4978.0> terminating
** (MatchError) no match of right hand side value: nil
    (membrane_core 0.5.2) lib/membrane/core/timer.ex:54: Membrane.Core.Timer.tick/1
    (elixir 1.10.3) lib/kernel.ex:2351: anonymous fn/2 in Kernel.update_in/3
    (elixir 1.10.3) lib/map.ex:837: Map.get_and_update/3
    (elixir 1.10.3) lib/access.ex:346: Access.get_and_update/3
    (elixir 1.10.3) lib/kernel.ex:2351: Kernel.update_in/3
    (membrane_core 0.5.2) lib/membrane/core/element/timer_controller.ex:66: Membrane.Core.Element.TimerController.handle_tick/2
    (membrane_core 0.5.2) lib/membrane/core/element/message_dispatcher.ex:36: Membrane.Core.Element.MessageDispatcher.handle_message/3
    (stdlib 3.12.1) gen_server.erl:637: :gen_server.try_dispatch/4
    (stdlib 3.12.1) gen_server.erl:711: :gen_server.handle_msg/6
    (stdlib 3.12.1) proc_lib.erl:249: :proc_lib.init_p_do_apply/3
Last message: {:EXIT, #PID<0.4941.0>, {{:badmatch, nil}, [{Membrane.Core.Timer, :tick, 1, [file: 'lib/membrane/core/timer.ex', line: 54]}, {Kernel, :"-update_in/3-fun-0-", 2, [file: 'lib/kernel.ex', line: 2351]}, {Map, :get_and_update, 3, [file: 'lib/map.ex', line: 837]}, {Access, :get_and_update, 3, [file: 'lib/access.ex', line: 346]}, {Kernel, :update_in, 3, [file: 'lib/kernel.ex', line: 2351]}, {Membrane.Core.Element.TimerController, :handle_tick, 2, [file: 'lib/membrane/core/element/timer_controller.ex', line: 66]}, {Membrane.Core.Element.MessageDispatcher, :handle_message, 3, [file: 'lib/membrane/core/element/message_dispatcher.ex', line: 36]}, {:gen_server, :try_dispatch, 4, [file: 'gen_server.erl', line: 637]}, {:gen_server, :handle_msg, 6, [file: 'gen_server.erl', line: 711]}, {:proc_lib, :init_p_do_apply, 3, [file: 'proc_lib.erl', line: 249]}]}}
State: {:state, {#PID<0.4978.0>, Membrane.Protocol.RTSP.Session.Container}, :one_for_one, {[Membrane.Protocol.RTSP.Transport, Membrane.Protocol.RTSP.Session.Manager], %{Membrane.Protocol.RTSP.Session.Manager => {:child, #PID<0.4979.0>, Membrane.Protocol.RTSP.Session.Manager, {Membrane.Protocol.RTSP.Session.Manager, :start_link, [%Membrane.Protocol.RTSP.Transport{key: {:via, Registry, {TransportRegistry, #Reference<0.189328267.2396520449.243971>}}, module: Membrane.Protocol.RTSP.Transport.TCPSocket}, %URI{authority: "REDACTED@192.168.1.137:554", fragment: nil, host: "192.168.1.137", path: "/axis-media/media.amp/", port: 554, query: nil, scheme: "rtsp", userinfo: "REDACTED"}, []]}, :permanent, 5000, :worker, [Membrane.Protocol.RTSP.Session.Manager]}, Membrane.Protocol.RTSP.Transport => {:child, #PID<0.4980.0>, Membrane.Protocol.RTSP.Transport, {Membrane.Protocol.RTSP.Transport, :start_link, [%Membrane.Protocol.RTSP.Transport{key: {:via, Registry, {TransportRegistry, #Reference<0.189328267.2396520449.243971>}}, module: Membrane.Protocol.RTSP.Transport.TCPSocket}, %URI{authority: "REDACTED@192.168.1.137:554", fragment: nil, host: "192.168.1.137", path: "/axis-media/media.amp/", port: 554, query: nil, scheme: "rtsp", userinfo: "REDACTED"}, []]}, :permanent, 5000, :worker, [Membrane.Protocol.RTSP.Transport]}}}, :undefined, 3, 5, [], 0, Membrane.Protocol.RTSP.Session.Container, [%Membrane.Protocol.RTSP.Transport{key: {:via, Registry, {TransportRegistry, #Reference<0.189328267.2396520449.243971>}}, module: Membrane.Protocol.RTSP.Transport.TCPSocket}, %URI{authority: "REDACTED", fragment: nil, host: "192.168.1.137", path: "/axis-media/media.amp/", port: 554, query: nil, scheme: "rtsp", userinfo: "REDACTED"}, []]}

```

**Comment by ConnorRigby** [comment]
After a bit of investigation, this seems to be a race condition.

**Comment by ConnorRigby** [comment]
After a bit more investigation, i think that this can only happen if i cancel a timer during the `tick` callback.

**Comment by ConnorRigby** [comment]
Current work around is to use `timer_interval` instead of `stop_timer`

**Comment by mat-hek** [comment]
Thanks for the catch, fix published as v0.5.3
